src/compile/data/bin.ts

```diff
@@ -19,36 +19,42 @@ function numberFormatExpr(expr: string, format: string) {
 function parse(model: Model): Dict<VgTransform[]> {
   return model.reduceFieldDef(function(binComponent: Dict<VgTransform[]>, fieldDef: FieldDef, channel: Channel) {
     const bin = model.fieldDef(channel).bin;
+    // Make this bin variable always an object
```

Comment:
The comment here is obviously left from our prior discussion. 

Your code is still incorrect when you apply bin to two channels, and each of them gets different auto `maxbins` parameters.  In that case, your code should produce two separate bin transform since the parameter is different. 

You should add a new test case to test this case.